server/src/main/java/org/opensearch/cluster/routing/WRRWeight.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Entity for Weighted Round Robin weights
+ *
+ * @opensearch.internal
+ */
+public class WRRWeight implements Writeable {
```

Comment:
nit : should we call it `WRRWeights` as it include weights for all attribute values.

Reply:
Sure. Will make this change